devutils/src/devutils/invoke_utils.py

```diff
@@ -390,30 +445,63 @@ def make_release(ctx):
             echo=True,
         )
 
-        print(f"Trigger the release with: git push origin {current_tag}")
+        print(f"Pushing tag: {current_tag}")
+        run(ctx, f"git push origin {current_tag}")
 
     @task
     def check_tag_version(ctx):
         """
         Checks if the current tag matches the latest version (exits with 1 if it
         does not match and with 0 if it does match).
         """
-        import importlib
-
-        mod = importlib.import_module(package_name)
+        module_version = poetry(
+            ctx,
+            f"run python -c 'import {package_name}; print({package_name}.__version__)'",
+        ).stdout.strip()
 
         tag = get_tag(tag_prefix)
         version = tag[tag.rfind("-") + 1 :]
 
-        if mod.__version__ == version:
+        if module_version == version:
             sys.stderr.write(f"Version matches ({version}) (exit(0))\n")
             sys.exit(0)
         else:
             sys.stderr.write(
-                f"Version does not match ({tag_prefix}: {mod.__version__} != repo tag: {version}).\nTags:{get_all_tags(tag_prefix)}\n(exit(1))\n"
+                f"Version does not match ({tag_prefix}: {module_version} != repo tag: {version}).\nTags:{get_all_tags(tag_prefix)}\n(exit(1))\n"
             )
             sys.exit(1)
 
+    def update_changelog_file(file: Path, version: str):
+        """Update the changelog file with the new version and changes"""
+
+        with open(file, "r+") as stream:
+            content = stream.read()
+
+            # Prepend a new section with the new version and changes
+            new_version_section = (
+                f"## {version} - {datetime.today().strftime('%Y-%m-%d')}\n\n"
+            )
+
+            changelog_start = re.search(r"# Changelog", content).end()
```

Comment:
What's the expected format of the changelog that this will act upon?

The format I was usually working on so far was:

```
# Changelog

## Unreleased

- xxx
- yyy

## 0.0.20 - 2024-01-31

- zzz
```

But then I'd expect this code here to just change the 'Unreleased' to the version here, but this isn't what the code is doing...

Reply:
I think it may be nicer to change the code to deal with that structure and raise an error if `## Unreleased` is not found right after `## Changelog` complaining that the changelog is not up to date.

Reply:
I don't have a strong opinion here either. We can add back the `Unreleased` header or just add the comments above the latest version. But we need to decide and be consistent.